src/pages/messages/components/NviCandidateActions.tsx

```diff
@@ -233,30 +243,85 @@ export const NviCandidateActions = ({ nviCandidate, nviCandidateQueryKey }: NviC
               {t('tasks.nvi.reject_nvi_candidate')}
             </Button>
 
-            <Divider sx={{ mb: '1rem' }} />
+            <RejectionDialog
+              open={hasSelectedRejectCandidate}
+              onCancel={() => setHasSelectedRejectCandidate(false)}
+              onAccept={async (reason) => {
+                await statusMutation.mutateAsync({ status: 'Rejected', reason });
+                setHasSelectedRejectCandidate(false);
+              }}
+              isLoading={statusMutation.isLoading}
+            />
 
-            {hasSelectedRejectCandidate && (
-              <MessageForm
-                confirmAction={async (reason) => {
-                  await statusMutation.mutateAsync({ status: 'Rejected', reason });
-                  setHasSelectedRejectCandidate(false);
-                }}
-                cancelAction={() => setHasSelectedRejectCandidate(false)}
-                fieldLabel={t('tasks.nvi.reject_nvi_candidate_form_label')}
-                buttonTitle={t('tasks.nvi.reject_nvi_candidate')}
-              />
-            )}
+            <Divider sx={{ mb: '1rem' }} />
           </>
         )}
 
-        {!hasSelectedRejectCandidate && (
-          <MessageForm
-            confirmAction={async (text) => await createNoteMutation.mutateAsync({ text })}
-            fieldLabel={t('tasks.nvi.note')}
-            buttonTitle={t('tasks.nvi.save_note')}
-          />
-        )}
+        <MessageForm
+          confirmAction={async (text) => await createNoteMutation.mutateAsync({ text })}
+          fieldLabel={t('tasks.nvi.note')}
+          buttonTitle={t('tasks.nvi.save_note')}
+        />
       </Box>
     </>
   );
 };
+
+interface RejectionDialogProps {
+  open: boolean;
+  onCancel: () => void;
+  onAccept: (reason: string) => Promise<unknown> | void;
+  isLoading: boolean;
+}
+
+const RejectionDialog = ({ open, onCancel, onAccept, isLoading }: RejectionDialogProps) => {
+  const { t } = useTranslation();
+  const [reason, setReason] = useState('');
+  const maxLength = 160;
+
+  const handleKeypress = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === 'Enter' || event.key === 'NumpadEnter') {
+      event.preventDefault();
+      onAccept(reason.trim());
+    }
+  };
+
+  return (
+    <Dialog open={open} onKeyDown={handleKeypress}>
```

Comment:
Naturlig å legge dette i en <form> i stedet for en custom `onKeyDown`? I praksis er det vel bare en form med ett teksfelt som skal submittes her
```suggestion
    <Dialog open={open}>
```

Reply:
Mac (safari) lukker vel også dialog med Esc vil jeg tro? https://mui.com/material-ui/react-dialog/
Når jeg åpner en dialog med en form ville jeg forventa at Esc lukka dialogen, mens enter blir en submit. 
Tenker fiksen her blir å bruke en form (eks. uten MUI og Formik: https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/close#examples)

Reply:
Mulig GitHub toloket det du skrev som kode😅 I took a calcuated risk, but man am I bad ar math. Skal fikse